starknet_py/tests/e2e/account/account_test.py

```diff
@@ -0,0 +1,179 @@
+        .prepare("1234", max_fee=0)
+        .estimate_fee(block_hash="latest")
+    )
+
+    assert estimated_fee.overall_fee > 0
+    assert (
+        estimated_fee.gas_price * estimated_fee.gas_usage == estimated_fee.overall_fee
+    )
+
+
+@pytest.mark.asyncio
+async def test_estimate_fee_for_declare_transaction(account, map_compiled_contract):
+    declare_tx = await account.sign_declare_transaction(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE
+    )
+
+    estimated_fee = await account.client.estimate_fee(tx=declare_tx)
+
+    assert isinstance(estimated_fee.overall_fee, int)
+    assert estimated_fee.overall_fee > 0
+    assert (
+        estimated_fee.gas_usage * estimated_fee.gas_price == estimated_fee.overall_fee
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("key, val", [(20, 20), (30, 30)])
+async def test_sending_multicall(account, map_contract, key, val):
+    calls = [
+        map_contract.functions["put"].prepare(key=10, value=10),
+        map_contract.functions["put"].prepare(key=key, value=val),
+    ]
+
+    res = await account.execute(calls=calls, max_fee=int(1e20))
+    await account.client.wait_for_tx(res.transaction_hash)
+
+    (value,) = await map_contract.functions["get"].call(key=key)
+
+    assert value == val
+
+
+@pytest.mark.run_on_devnet
+@pytest.mark.asyncio
+async def test_get_block_traces(gateway_account):
+    traces = await gateway_account.client.get_block_traces(block_number=2)
+
+    assert traces.traces != []
+
+
+@pytest.mark.asyncio
+async def test_rejection_reason_in_transaction_receipt(account, map_contract):
+    res = await map_contract.functions["put"].invoke(key=10, value=20, max_fee=1)
+
+    with pytest.raises(TransactionRejectedError):
+        await account.client.wait_for_tx(res.hash)
+
+    transaction_receipt = await account.client.get_transaction_receipt(res.hash)
+
+    if isinstance(account.client, GatewayClient):
+        assert "Actual fee exceeded max fee." in transaction_receipt.rejection_reason
+
+
+@pytest.mark.asyncio
+async def test_sign_and_verify_offchain_message_fail(account, typed_data):
+    signature = account.sign_message(typed_data)
+    signature = [signature[0] + 1, signature[1]]
+    result = await account.verify_message(typed_data, signature)
+
+    assert result is False
+
+
+@pytest.mark.asyncio
+async def test_sign_and_verify_offchain_message(account, typed_data):
+    signature = account.sign_message(typed_data)
+    result = await account.verify_message(typed_data, signature)
+
+    assert result is True
+
+
+@pytest.mark.asyncio
+async def test_get_class_hash_at(map_contract, account):
+    class_hash = await account.client.get_class_hash_at(
+        map_contract.address, block_hash="latest"
+    )
+
+    assert class_hash != 0
+
+
+@pytest.mark.asyncio()
+async def test_get_nonce(gateway_account):
+    nonce = await gateway_account.get_nonce()
+
+    assert nonce >= 0  # TODO maybe better test
```

Comment:
👀

Reply:
I wonder if I should change this test?

Reply:
I don't have any idea to test it any better while staying simple, so I'd say it's ok as it is.

Reply:
We do 😎 @cptartur will add it soon